projects/pgai/pgai/configuration.py

```diff
@@ -0,0 +1,361 @@
+import textwrap
+from dataclasses import dataclass, fields
+from typing import Any, Literal, Protocol, runtime_checkable
+
+from alembic.autogenerate.api import AutogenContext
+
+from pgai.vectorizer.processing import ProcessingDefault
+
+
+@runtime_checkable
+class SQLArgumentProvider(Protocol):
+    def to_sql_argument(self) -> str: ...
+
+
+@runtime_checkable
+class PythonArgProvider(Protocol):
+    def to_python_arg(self) -> str: ...
+
+
+def format_python_arg(config_type: str, instance: Any) -> str:
```

Comment:
can't we right a similar function for sql to make all this simpler to write? I mean all parameters can be passed in with {name} => {value} and the way to render the value is type dependent, which we can switch on. We can also skip any None's etc